BTrees/_base.py

```diff
@@ -815,9 +815,7 @@ def __delitem__(self, key):
         self._del(self._to_key(key))
 
     def clear(self):
-        if self._data:
-            # In the case of __init__, this was already set by __new__
-            self._data = []
+        self._data = []
```

Comment:
I'm not agains for having `__setstate__` setting `._data` itself directly, but the code outline with `clear()` was already there and I've tried to only fix things up. Regarding pure-python codepaths my approach is only try to not to break it because here at Nexedi we only use original C-based codes.

In any case, be it separate `.clear()` or setting `._data` directly in `__setstate__()`, I would uncoditinally initialize `._data` to [] until there is a clear performance win in not doing so.

> If clear is called by __setstate__ and during steady state, that subclass will now be broken, when it wasn't before.

Could you please clarify? current state is `__setstate__` is already calling `clear`. Why it was not broken before then?